pyquil/api/_rewrite_arithmetic.py

```diff
@@ -138,7 +138,7 @@ def expr_mref(expr: object) -> MemoryReference:
                 # so we divide by 8...
                 expr = str(Div(inst.scale, 8))
                 updated.inst(SetScale(inst.frame, expr_mref(expr)))
-        else:
+        elif not isinstance(inst, Declare):
```

Comment:
Maybe a note here on why `Declare` is special?

Reply:
Done!